arch/platform/openmote/dev/at86rf215-arch.c

```diff
@@ -0,0 +1,169 @@
+#define AT86RF215_IRQ_PIN_MASK        GPIO_PIN_MASK(AT86RF215_IRQ_PIN)
+/*---------------------------------------------------------------------------*/
+extern void at86rf215_isr(void);
+/*---------------------------------------------------------------------------*/
+static void
+at86rf215_interrupt_handler(gpio_hal_pin_mask_t pin_mask)
+{
+    at86rf215_isr();
+}
+/*---------------------------------------------------------------------------*/
+static gpio_hal_event_handler_t irq_handler = {
+  .next = NULL,
+  .handler = at86rf215_interrupt_handler,
+  .pin_mask = gpio_hal_pin_to_mask(AT86RF215_IRQ_PIN) << (AT86RF215_IRQ_PORT << 3),
+};
+/*---------------------------------------------------------------------------*/
+void 
+at86rf215_arch_enable_EXTI(void)
+{
+    GPIO_ENABLE_INTERRUPT(AT86RF215_IRQ_PORT_BASE, AT86RF215_IRQ_PIN_MASK);
+    NVIC_EnableIRQ(AT86RF215_GPIOx_VECTOR);
+}
+/*---------------------------------------------------------------------------*/
+void
+at86rf215_arch_disable_EXTI(void)
+{
+    GPIO_DISABLE_INTERRUPT(AT86RF215_IRQ_PORT_BASE, AT86RF215_IRQ_PIN_MASK);
+}
+/*---------------------------------------------------------------------------*/
+void
+at86rf215_arch_set_RSTN(void)
+{   
+    GPIO_CLR_PIN(AT86RF215_RSTN_PORT_BASE, AT86RF215_RSTN_PIN_MASK);
+}
+/*---------------------------------------------------------------------------*/
+void
+at86rf215_arch_clear_RSTN(void)
+{
+    GPIO_SET_PIN(AT86RF215_RSTN_PORT_BASE, AT86RF215_RSTN_PIN_MASK);
+}
+/*---------------------------------------------------------------------------*/
+void
+at86rf215_arch_spi_select(void)
+{   
+    GPIO_CLR_PIN(AT86RF215_SPI_CSN_PORT_BASE, AT86RF215_SPI_CSN_PIN_MASK);
+}
+/*---------------------------------------------------------------------------*/
+void
+at86rf215_arch_spi_deselect(void)
+{
+    GPIO_SET_PIN(AT86RF215_SPI_CSN_PORT_BASE, AT86RF215_SPI_CSN_PIN_MASK);
+}
+/*---------------------------------------------------------------------------*/
+uint8_t 
+at86rf215_arch_spi_txrx(uint8_t b)
+{
+    SPIX_WAITFORTxREADY(AT86RF215_SPI_INSTANCE);
+    SPIX_BUF(AT86RF215_SPI_INSTANCE) = b;
+    SPIX_WAITFOREOTx(AT86RF215_SPI_INSTANCE);
+    SPIX_WAITFOREORx(AT86RF215_SPI_INSTANCE);
+    b = SPIX_BUF(AT86RF215_SPI_INSTANCE);
+    return b;
+}
+/*---------------------------------------------------------------------------*/
+void
+at86rf215_arch_init(void)
+{
+    /* Initialize power pin */
+    GPIO_SOFTWARE_CONTROL(AT86RF215_PWR_PORT_BASE , AT86RF215_PWR_PIN_MASK );
+    GPIO_SET_OUTPUT(AT86RF215_PWR_PORT_BASE , AT86RF215_PWR_PIN_MASK );
+    
+    /* Initialize reset pin */
+    GPIO_SOFTWARE_CONTROL(AT86RF215_RSTN_PORT_BASE ,AT86RF215_RSTN_PIN_MASK );
+    GPIO_SET_OUTPUT(AT86RF215_RSTN_PORT_BASE ,AT86RF215_RSTN_PIN_MASK );
+
+    /* Turn the radio off */
+    GPIO_CLR_PIN(AT86RF215_PWR_PORT_BASE , AT86RF215_PWR_PIN_MASK );
+    GPIO_CLR_PIN(AT86RF215_RSTN_PORT_BASE ,AT86RF215_RSTN_PIN_MASK );
+    for(uint16_t delay=0; delay<0xA2C2; delay++);
```

Comment:
The compiler might optimize these away, there should be some delay function to call if that is what is required.